wikimedia/wikipedia-ios
# fix crash in SearchViewController

**Phabricator:** N/A, but follow up from https://phabricator.wikimedia.org/T208892

### Notes
* I believe we're using the wrong check here. We want to make sure there are some search results (otherwise we should be allowing deletes, as the cell is a historical search term), but we need to make sure the index is smaller than the search results, not the number of saved historical search terms.
* This was crashing on device but not on sim b/c my sims had fewer historical saved searches.

### Test Steps
1. Search screen - ensure deleting old search terms work. 
2. Make several searches.
2. Ensure swiping right on search results works
3. Cancel the search, ensure deleting old search terms still works.
